### PR TITLE
Simplify some Common ActionRunnable Uses

### PR DESCRIPTION
**Pull request by original-brownbear**
Especially in the snapshot code there's a lot
of logic chaining `ActionRunnables` in tricky
ways now and the code is getting hard to follow.
This change introduces two convinience methods that
make it clear that a wrapped listener is invoked with
certainty in some trickier spots and shortens the code a bit.

### CODE REVIEW

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by original-brownbear** [comment]
Thanks Ryan!
